Tests/SwiftSyntaxBuilderTest/StructTests.swift

```diff
@@ -55,4 +49,56 @@ final class StructTests: XCTestCase {
     }
     """)
   }
+
+  func testEmptyStructWithBuilder() {
+    let leadingTrivia = Trivia.garbageText("␣")
+    let test = StructDecl(identifier: "TestStruct", members: MemberDeclBlock())
+
+    let syntax = test.buildSyntax(format: Format(), leadingTrivia: leadingTrivia)
+
+    var text = ""
+    syntax.write(to: &text)
+
+    XCTAssertEqual(text, """
```

Comment:
This test fails and generates
```
␣struct TestStruct{
}
```

We need a leading space before the `{`.
We could add `requires_leading_space `on [LeftBrace](https://github.com/apple/swift/blob/main/utils/gyb_syntax_support/Token.py#L217).

Reply:
Requiring a leading space for left brace sounds reasonable to me, but I’m not too upset about the current behavior either – it compiles, everything else just just best effort and we’ll never match a proper formatter in that.

Could you do that change in a follow-up PR and just leave it as is for now?